File: eform-client/src/app/plugins/modules/time-planning-pn/models/pay-rule-sets/pay-rule-set-update.model.ts
import { PayDayRuleModel } from './pay-day-rule.model';
import { PayDayTypeRuleModel } from '../pay-day-type-rules/pay-day-type-rule.model';

export class PayRuleSetUpdateModel {
  // id is passed separately in the URL, not in the body
  name: string;
  payDayRules: PayDayRuleModel[];
  payDayTypeRules: PayDayTypeRuleModel[];
}

File: eFormAPI/Plugins/TimePlanning.Pn/TimePlanning.Pn/Infrastructure/Models/PayRuleSet/PayRuleSetUpdateModel.cs
/*
The MIT License (MIT)

Copyright (c) 2007 - 2021 Microting A/S
*/

namespace TimePlanning.Pn.Infrastructure.Models.PayRuleSet;

using System.Collections.Generic;
using PayDayTypeRule;

public class PayRuleSetUpdateModel
{
    // Id is passed in the URL route, not in the model body
    public string Name { get; set; }
    public List<PayDayRuleModel> PayDayRules { get; set; } = new List<PayDayRuleModel>();
    public List<PayDayTypeRuleModel> PayDayTypeRules { get; set; } = new List<PayDayTypeRuleModel>();
}